packages/react/src/components/Dialog/index.tsx

```diff
@@ -117,6 +118,7 @@ export default class Dialog extends React.Component<DialogProps, DialogState> {
     const Dialog = (
       <FocusTrap
         focusTrapOptions={{
+          allowOutsideClick: true,
```

Comment:
I spent way too much time here getting stuck, but apparently we weren't testing this correctly earlier since we were just checking that the dialog handler was invoked but not the prop function. Since we already have `ClickOutsideListener` we don't need to let focus trap handle it.